servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/Publisher.java

```diff
@@ -964,51 +962,45 @@ protected Publisher() {
 
     /**
      * Map each element of this {@link Publisher} into a {@link Single}&lt;{@link R}&gt; and flatten all signals
-     * emitted from each mapped {@link Single}&lt;{@link R}&gt; into the returned
-     * {@link Publisher}&lt;{@link R}&gt;.
+     * emitted from each mapped {@link Single}&lt;{@link R}&gt; into the returned {@link Publisher}&lt;{@link R}&gt;.
      * <p>
      * The behavior is the same as {@link #flatMapMergeSingle(Function, int)} with the exception that if any
      * {@link Single} returned by {@code mapper}, terminates with an error, the returned {@link Publisher} will not
      * immediately terminate. Instead, it will wait for this {@link Publisher} and all {@link Single}s to terminate and
-     * then terminate the returned {@link Publisher} with all errors emitted by the {@link Single}s produced by the
+     * then terminate the returned {@link Publisher} with errors emitted by the {@link Single}s produced by the
      * {@code mapper}.
      * <p>
      * To control the amount of concurrent processing done by this operator see
      * {@link #flatMapMergeSingleDelayError(Function, int)}.
      * <p>
-     * This method is similar to {@link #map(Function)} but the result is asynchronous, and provides a data
+     * This method is similar to {@link #map(Function)} but the result is asynchronous, unordered, and provides a data
      * transformation in sequential programming similar to:
      * <pre>{@code
-     *     Executor e = ...;
-     *     List<T> tResults = resultOfThisPublisher();
-     *     List<R> rResults = ...; // assume this is thread safe
-     *     List<Throwable> errors = ...;  // assume this is thread safe
-     *     CountDownLatch latch =  new CountDownLatch(tResults.size());
-     *     for (T t : tResults) {
+     *     ExecutorService e = ...;
+     *     List<Future<R>> futures = ...; // assume this is thread safe
+     *     for (T t : resultOfThisPublisher()) {
      *         // Note that flatMap process results in parallel.
-     *         e.execute(() -> {
-     *             try {
-     *                 R r = mapper.apply(t); // Asynchronous result is flatten into a value by this operator.
-     *                 rResults.add(r);
-     *             } catch (Throwable cause) {
-     *                 errors.add(cause);  // Asynchronous error is flatten into an error by this operator.
-     *             } finally {
-     *                 latch.countdown();
-     *             }
-     *         });
+     *         futures.add(e.submit(() -> {
+     *             processResult(mapper.apply(t));
+     *             return null;
+     *         }));
      *     }
-     *     latch.await();
-     *     if (errors.isEmpty()) {
-     *         return rResults;
+     *     List<Throwable> errors = ...;
+     *     for (Future<R> future : futures) {
+     *         try {
+     *           future.get(); // Throws if the processing for this item failed.
```

Comment:
Because merge is unordered, consider using [CompletableFuture.allOf(...)](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#allOf-java.util.concurrent.CompletableFuture...-) to demonstrate that.

Reply:
`delayError` error propagation is different than `CompletableFuture.allOf(..)` because it may potentially return all exceptions (e.g. not bail on the first one).